src/ConnectedMode/ConnectionInfo.cs

```
@@ -32,19 +31,33 @@ public enum ConnectionServerType
 /// <summary>
 /// Model containing connection information, intended to be used by UI components
 /// </summary>
-/// <param name="Id">The organization key for SonarCloud or the server uri for SonarQube</param>
-/// <param name="ServerType">The type of server (SonarCloud, SonarQube)</param>
-public record ConnectionInfo(string Id, ConnectionServerType ServerType)
+public record ConnectionInfo
 {
-    public static ConnectionInfo From(ServerConnection serverConnection)
+    /// <summary>
+    /// Model containing connection information, intended to be used by UI components
+    /// </summary>
+    /// <param name="id">The organization key for SonarCloud or the server uri for SonarQube</param>
+    /// <param name="serverType">The type of server (SonarCloud, SonarQube)</param>
+    /// <param name="cloudServerRegion"> Only applicable to <see cref="ConnectionServerType.SonarCloud"/>. If not provided, defaults to <see cref="Core.Binding.CloudServerRegion.Eu"/>
+    /// It is ignored for <see cref="ConnectionServerType.SonarQube"/> </param>
+    public ConnectionInfo(string id, ConnectionServerType serverType, CloudServerRegion cloudServerRegion = null)
     {
-        return serverConnection switch
+        Id = id;
+        ServerType = serverType;
+        CloudServerRegion = serverType == ConnectionServerType.SonarCloud ? (cloudServerRegion ?? CloudServerRegion.Eu) : null;
```

Maybe we should throw instead of defaulting to null here?

---

We are defaulting to null only for SonarQube.

---

Sorry, I meant throw IF it's set for SonarQube